### ISSUE TITLE
Additional functionality request 

### ISSUE THREAD

**Issue by abelmokadem** [hvac/hvac#716]
Hi all, we have a library that is an extension of hvac and I'm wondering if there is room to move some, maybe all of the code into hvac? It is considered "higher level" functionality over the api. Anyone here to discuss this? Link to code: https://github.com/schubergphilis/hashivaultlib/blob/master/hashivaultlib/hashivaultlib.py

**Comment by Tylerlhess** [comment]
I am in no way a maintainer or really have any say in hvac's development direction I have just had a few pulls to resolve some issues I have personally come across.

I looked over the code and had some questions, quick thoughts, and my tl;dr; on the functionality.

You are correct about this being a second layer solution but I do see some functionality that might be nice to haves and thus why it was created in the first place. The main issue is that you have a file right now and I'm sure it works but there are no recorded test cases or automated install files. (Not sure how much of the dependancies are covered by just having vault but you definitely should have a minimum hvac version.

I like the recursive secret pull. That is definitely a nice to have. The Token class looks awesome for those that are managing their own token life cycle but Vault Agent handles most of that natively. 

However, HVAC is, according to the README.md, the "Hashicorp Vault API client for Python 3.x". So I don't think these additional functions would fit within those parameters as they are not in the Vault API (as far as I know). 

tl;dr; I think recursive secret pulls is a great functional nice to have, but token management is obsoleted through vault agent. These functions aren't part of the API yet so until they become such I don't think they fit in the API client for python.

**Comment by briantist** [comment]
Hi @abelmokadem thanks for opening this! I took a look at the library and I think @Tylerlhess hit the nail on the head: this is more of a library with functionality that broadly aligns with the Vault API and its primitives, and not so much about higher level workflows that you can build on Vault.

If there are specific helper functions or other things in `hashivaultlib` that you think might be a good fit here, please let us know and let's discuss those individually, or as groups of functionality.

I would recommend starting that as a [Discussion topic](https://github.com/hvac/hvac/discussions/categories/ideas) and we can go from there!
